### DISCUSSION TITLE
bug: globalThis is not defined

### DISCUSSION THREAD

**Discussion by aerodynamics-digital** [ionic-team/capacitor#3871] [Q&A]
### Capacitor Version
Latest Dependencies:

  @capacitor/cli: 2.4.3
  @capacitor/core: 2.4.3
  @capacitor/android: 2.4.3
  @capacitor/electron: 2.4.3
  @capacitor/ios: 2.4.3

Installed Dependencies:

  @capacitor/cli 2.4.3
  @capacitor/core 2.4.3
  @capacitor/ios 2.4.3
  @capacitor/android 2.4.3
  @capacitor/electron not installed

```
PASTE OUTPUT HERE
```

### Platform(s)
android

### Current Behavior
in the emulator i get the error: E/Capacitor/Console: File: capacitor-runtime.js - Line 361 - Msg: ReferenceError: globalThis is not defined

in the chrome inspector i can see more details about it: ReferenceError: globalThis is not defined

### Expected Behavior
app should load correctly

### Code Reproduction
ionic app with angular 11 and angularfire 6.1.1 installed (but any other service that caches on globalThis will get this error)

### Other Technical Details

`npm --version` output:
6.14.9

`node --version` output:
v12.15.0

`pod --version` output (iOS issues only):

**Comment by rmakash** [comment]
Hi,
I faced the same problem after updating @angular/fire to the latest version!
I downgraded @angular/fire to version 6.0.4 and all worked fine.
Hope this helps.

  @capacitor/cli: 2.4.3
  @capacitor/core: 2.4.3
  @capacitor/android: 2.4.3
  @capacitor/electron: 2.4.3
  @capacitor/ios: 2.4.3

Installed Dependencies:

  @capacitor/cli 2.4.3
  @capacitor/core 2.4.3
  @capacitor/ios 2.4.3
  @capacitor/android 2.4.3
  @capacitor/electron not installed

npm --version
6.14.9
node --version
v14.13.1

**Comment by aerodynamics-digital** [comment]
yes i know, we would like to use the new functionalities added in 6.1, does anyone have a workaround? i tried adding it in polyfills but i could'd get it to work

**Comment by jcesarmobile** [comment]
this is not a Capacitor issue, moving to discussions so maybe somebody can help you

**Comment by aparajita** [comment]
This is not a Capacitor issue, `globalThis` is only supported by recent versions of the webview. The problem lies with @angular/fire.

**Comment by samipshresthaq** [comment]
You can import globalThis as polyfill using this package https://www.npmjs.com/package/@ungap/global-this.
Ref: https://zenn.dev/rdlabo/articles/b25c18b243a5b8

I fixed my issue using this polyfill.

**Comment by ghost** [comment]
Hello,

Install globalthis:
`npm install globalthis `

Add:
`import 'globalthis/auto';`
in polyfills.ts

**Reply by daveshirman** [reply]
Outstanding. Thank you.

**Reply by ram-xv** [reply]
Thank you 👏

**Reply by danibetzamora** [reply]
Thank you so much!

**Reply by Gautammer** [reply]
**_Thank you very much, You saved life ✌️_**

**Reply by cjchrist777** [reply]
Thanks bro!!! This is the fix for ionic 7 android white screen issue

**Reply by sachithd** [reply]
Thanks very much.

**Reply by divyeshramani** [reply]
I am getting same error with ionic 7 react on android.  Where should polyfills.ts file go?  I put in root folder but didn't work.
